src/order/Order.java

```diff
@@ -7,20 +7,22 @@
 import user.User;
 
 import java.util.ArrayList;
-import java.util.List;
+import java.util.UUID;
 
 public class Order implements Comparable<Order>{
+    private UUID orderId;
     private User client;
     private Restaurant restaurant;
-    private List<MenuItem> orderItems;
+    private ArrayList<MenuItem> orderItems;
```

Comment:
please change every collection to have as reference type the interface and for the real object a concrete implementation of that interface
ex: List<MenuItem> list = new Arraylist<>()